test/instances/morpho-blue/MorphoBlue_wstETH_USDC_Hyperdrive.t.sol

```diff
@@ -0,0 +1,818 @@
+        (
+            uint256 totalSupplyAssetsBefore,
+            uint256 totalSupplySharesBefore
+        ) = getSupply();
+        AccountBalances memory bobBalancesBefore = getAccountBalances(bob);
+        AccountBalances memory hyperdriveBalancesBefore = getAccountBalances(
+            address(hyperdrive)
+        );
+
+        // Bob removes his liquidity with base as the target asset.
+        (uint256 baseProceeds, uint256 withdrawalShares) = removeLiquidity(
+            bob,
+            lpShares
+        );
+        assertEq(withdrawalShares, 0);
+
+        // Bob should receive approximately as much base as he contributed since
+        // no time as passed and the fees are zero.
+        assertApproxEqAbs(baseProceeds, _contribution, 1e3);
+
+        // Ensure that the withdrawal was processed as expected.
+        verifyWithdrawal(
+            bob,
+            baseProceeds,
+            true,
+            totalSupplyAssetsBefore,
+            totalSupplySharesBefore,
+            bobBalancesBefore,
+            hyperdriveBalancesBefore
+        );
+    }
+
+    function test_round_trip_lp_withdrawal_shares(
+        uint256 _contribution,
+        uint256 _variableRate
+    ) external {
+        // Bob adds liquidity with base.
+        _contribution = _contribution.normalizeToRange(100e6, 100_000e6);
+        IERC20(hyperdrive.baseToken()).approve(
+            address(hyperdrive),
+            _contribution
+        );
+        uint256 lpShares = addLiquidity(bob, _contribution);
+
+        // Alice opens a large short.
+        vm.stopPrank();
+        vm.startPrank(alice);
+        uint256 shortAmount = hyperdrive.calculateMaxShort();
+        IERC20(hyperdrive.baseToken()).approve(
+            address(hyperdrive),
+            shortAmount
+        );
+        openShort(alice, shortAmount);
+
+        // Bob removes his liquidity with base as the target asset.
+        (uint256 baseProceeds, uint256 withdrawalShares) = removeLiquidity(
+            bob,
+            lpShares
+        );
+        assertGt(withdrawalShares, 0);
+
+        // The term passes and interest accrues.
+        _variableRate = _variableRate.normalizeToRange(0, 2.5e18);
+        advanceTime(POSITION_DURATION, int256(_variableRate));
+        hyperdrive.checkpoint(hyperdrive.latestCheckpoint(), 0);
+
+        // Bob should be able to redeem all of his withdrawal shares for
+        // approximately the LP share price.
+        uint256 lpSharePrice = hyperdrive.getPoolInfo().lpSharePrice;
+        uint256 withdrawalSharesRedeemed;
+        (baseProceeds, withdrawalSharesRedeemed) = redeemWithdrawalShares(
+            bob,
+            withdrawalShares
+        );
+        assertEq(withdrawalSharesRedeemed, withdrawalShares);
+
+        // Bob should receive base approximately equal in value to his present
+        // value.
+        assertApproxEqAbs(
+            baseProceeds,
+            withdrawalShares.mulDown(lpSharePrice),
+            1e3
```

Comment:
okay so this must have been one of the tolerances you had to tweak for a 6 decimal place USDC token. I believe this was 1e12 for the other tokens (which obviously wouldn't work here); however, an error of 12/18 > an error of 3/6 so this seems acceptable proportionally

Reply:
I'd say that an error of 1e12 is better than 1e3 since 1e3 only leaves 3 decimals of precision and 1e12 leaves 6. That said, There are some confounding variables caused by using `mulDown` instead of `mulDivDown` with the present value and the lp total supply. I'll try to tighten the bound.

Reply:
I was able to drop this to `100`.